src/components/clusters/wizards/common/ClusterSettings/MachinePool/ComputeNodeCount/ComputeNodeCount.test.tsx

```diff
@@ -38,15 +57,56 @@ const renderWithFormik = (props = {}) =>
   );
 
 describe('ComputeNodeCount', () => {
+  const mockSetFieldValue = jest.fn();
+  const mockValidateField = jest.fn();
+
+  const createMockFormState = (overrides = {}) => {
+    const values: Record<string, any> = {
+      [FieldId.Hypershift]: 'true',
+      [FieldId.MachinePoolsSubnets]: [],
+      [FieldId.NodesCompute]: 2,
+      [FieldId.MachineType]: 'm5.xlarge',
+      [FieldId.CloudProviderId]: 'aws',
+      [FieldId.Product]: 'ROSA',
+      [FieldId.BillingModel]: 'marketplace',
+      [FieldId.Byoc]: 'true',
+      [FieldId.MultiAz]: 'false',
+      ...overrides,
+    };
+    return {
+      values,
+      getFieldProps: (name: string) => ({
+        value: values[name] ?? '',
+        onChange: jest.fn(),
+        onBlur: jest.fn(),
+        name,
+      }),
+      getFieldMeta: () => ({ touched: false, error: undefined }),
+      setFieldValue: mockSetFieldValue,
+      validateField: mockValidateField,
+    };
+  };
+
   beforeEach(() => {
     jest.clearAllMocks();
-    (useFormStateModule.useFormState as jest.Mock).mockReturnValue({
-      values: {},
-      getFieldProps: () => ({ value: 2, onChange: jest.fn() }),
-      getFieldMeta: () => ({ touched: true, error: undefined }),
-      setFieldValue: jest.fn(),
-      validateField: jest.fn(),
+    mockSetFieldValue.mockClear();
+    mockValidateField.mockClear();
+
+    (useFormStateModule.useFormState as jest.Mock).mockReturnValue(createMockFormState());
```

Comment:
This is all so messy.  I am guessing this is AI.  
`jest.mock('~/components/clusters/wizards/hooks/useFormState');`  --> at top of file
`const mockedUseFormState = useFormState;`  --> At top of test suite
` mockedUseFormState.mockReturnValue({ 
      values: { ...initialValues }, etc...})`  --> per test